api/v1alpha1/imageclusterinstall_types.go

```diff
@@ -34,14 +34,36 @@ const (
 	HostConfigurationSucceededMessage = "Configuration image is attached to the referenced host"
 )
 
-// ClusterConfigSpec defines the desired state of ClusterConfig
-type ClusterConfigSpec struct {
-	clusterinfo.ClusterInfo `json:",inline"`
+// ImageClusterInstallSpec defines the desired state of ImageClusterInstall
+type ImageClusterInstallSpec struct {
+	// ClusterDeploymentRef is a reference to the ClusterDeployment.
+	// +optional
+	ClusterDeploymentRef *corev1.LocalObjectReference `json:"clusterDeploymentRef"`
+
+	// ImageSetRef is a reference to a ClusterImageSet.
+	ImageSetRef hivev1.ClusterImageSetReference `json:"imageSetRef"`
+
+	// ClusterMetadata contains metadata information about the installed cluster.
+	// This must be set as soon as all the information is available.
+	// +optional
+	ClusterMetadata *hivev1.ClusterMetadata `json:"clusterMetadata"`
+
+	// Version is the target OCP version for the cluster
+	// TODO: should this use ImageSetRef?
+	Version string `json:"version,omitempty"`
+
+	// MasterIP is the desired IP for the host
+	MasterIP string `json:"masterIP,omitempty"`
 
-	// PullSecretRef is a reference to new cluster-wide pull secret.
-	// If defined, it will replace the secret located at openshift-config/pull-secret.
-	// The type of the secret must be kubernetes.io/dockerconfigjson.
-	PullSecretRef *corev1.LocalObjectReference `json:"pullSecretRef,omitempty"`
+	// Hostname is the desired hostname for the host
+	Hostname string `json:"hostname,omitempty"`
+
+	// ReleaseRegistry is the registry of the installed release image
+	// TODO: should this also use the registry of ImageSetRef?
+	ReleaseRegistry string `json:"releaseRegistry,omitempty"`
+
+	// ClusterID is the desired ID for the new cluster
+	ClusterID string `json:"clusterID,omitempty"`
```

Comment:
Maybe make a mutating webhook to generate this if it wasn't provided.

Reply:
LCA will do the generating if you leave this empty anyway, so you don't have to bother making a mutating webhook

Reply:
Removed this.